### DISCUSSION TITLE
as prop for Link component

### DISCUSSION THREAD

**Discussion by wenerme** [TanStack/router#141] [Ideas]
Something like this https://v5.reactrouter.com/web/api/Link/component-reactcomponent

Current Link force using `a`, but want to use the component from ui lib

```jsx
<Link as={<MyLink/>} />
<Link as={MyLink} />
```

Rendered like 

https://github.com/wenerme/wode/blob/main/packages/reaction/src/render/flexRender.ts#L24-L30

**Comment by wenerme** [comment]
better to wrap a custom component.

**Comment by helmturner** [comment]
I'll bump this. I'm not sure if `a` is the only dom binding, but if so this router could theoretically be implemented in React Native should that binding be made optional.
